release-22.2: logictest: fix some edge cases

yuzefovich opened the pull request:
Backport:
  * 1/1 commits from "logictest: use v22_2 schema in mixed version test" (#98695) (reduced version)
  * 1/1 commits from "logictest: correctly disable auto stats on system tables" (#99987)
  * 1/1 commits from "logictest: disable DistSQL while updating DistSQL span resolver" (#103397)

Please see individual PRs for details.

/cc @cockroachdb/release

Fixes: #105612.

Release justification: bug fix.

yuzefovich committed:
logictest: correctly disable auto stats on system tables

This commit fixes an oversight of 9c6fcd19a0892592b11bc803408b6daad0c780dc
where we attempted to disable the auto stats collection on system tables
by overriding the corresponding cluster setting before the cluster is
started. The problem with that change was that we overrode the cluster
setting object that is only used by the temp storage config and has no
bearing on the settings used by each node in the cluster. As it turns
out, previously we didn't explicitly initialize the `Settings` field of
the arguments for each node, so a fresh copy was always created. This
oversight is now fixed - we explicitly create a settings object for each
node / tenant and then override the auto stats on system tables to be
disabled on it.

Release note: None

yuzefovich committed:
logictest: disable DistSQL while updating DistSQL span resolver

Over the last half a year we made a couple of changes that changed which
session data internal queries are using, and as a result internal
queries can now use DistSQL in most places. This can now lead to rare
flakes in `fakedist-*` logic test configurations where we're updating
the DistSQL span resolver after having started the cluster, meaning that
internal queries that are using DistSQL might be running concurrently
with the span resolver update. This is not allowed by the interface, so
this commit should fix these rare flakes by explicitly disabling DistSQL
before the cluster is started in `fakedist-*` configs until after the
span resolver is updated, at which point the DistSQL setting is reset.

I believe this violation of the interface is the root cause for segfault
we saw in #100108, so this commit reverts the debugging introduced in
response to that ticket.

Additionally, previously we updated the span resolver only on a single
node in the cluster, and this commit updates it on all nodes in the
cluster, which seems more reasonable.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/105636)
<!-- Reviewable:end -->

yuzefovich committed:
logictest: remove local-v1.1-at-v1.0-noupgrade config

This commit is reduced version of 4424f8696e84dce1396da0675984d5b4b4fda645
which deleted `local-v1.1-at-v1.0-noupgrade` logic test config. This
config is not that useful and is in my way of backporting other fixes.

Release note: None
